master/buildbot/worker/ec2.py

```diff
@@ -523,6 +542,7 @@ def _request_spot_instance(self):
             raise LatentWorkerFailedToSubstantiate()
```

Comment:
Slowly working on it but there's a bit of a problem.

Looking at the existing tests for spot instances they all call `_start_instance()` which is fine except this method calls `self.ec2.create_instance` so the tests, unless I'm missing something obvious, basically don't cover spot instances at all and the existing coverage seems to confirm this:

* https://codecov.io/gh/buildbot/buildbot/src/master/master/buildbot/worker/ec2.py#L471
* https://codecov.io/gh/buildbot/buildbot/src/master/master/buildbot/worker/ec2.py#L491
* https://codecov.io/gh/buildbot/buildbot/src/master/master/buildbot/worker/ec2.py#L564

I did start to work on updating `test_start_spot_instance()` to call `_request_spot_instance()` but found out moto does not implement `describe_spot_price_history()` so we won't be able to get coverage for `_bid_price_from_spot_price_history()`. 

The other issue, which I'd certainly welcome some suggestions on is how to get moto to create a spot request then modify the state of the request so `_wait_for_request()` can return correctly. I tried digging around on moto's own tests and it looks like I might be able to do something like:

```python 
models = get_model('SpotInstanceRequest', 'us-east-1')
# loop over models in a thread and modify the state so _wait_for_request() returns
```

But I'm not sure that's the right approach (or if it will even work) so I'd certainly welcome some suggestions. If I can get `test_start_spot_instance()` working modifying the other tests shouldn't be too difficult which would be great because then spot requests will actually be covered in the tests.

Thoughts?